util/network-devp2p/src/ip_utils.rs

```diff
@@ -225,28 +225,17 @@ mod getinterfaces {
 				let sa: *const sockaddr_in = sa as *const sockaddr_in;
 				let sa = unsafe { &*sa };
 				let (addr, port) = (sa.sin_addr.s_addr, sa.sin_port);
-				(IpAddr::V4(Ipv4Addr::new(
-					(addr & 0x0000_00FF) as u8,
-					((addr & 0x0000_FF00) >>  8) as u8,
-					((addr & 0x00FF_0000) >> 16) as u8,
-					((addr & 0xFF00_0000) >> 24) as u8)),
-					port)
+				// convert u32 to an `Ipv4 address`, but the u32 must be converted to `host-order`
+				// that's why `from_be` is used!
+				(IpAddr::V4(Ipv4Addr::from(<u32>::from_be(addr))), port)
```

Comment:
This should now work both on `little endian` and `big-endian` machines!

Reply:
why wouldn't it before? The IPv6 code looks endian-reliant but simple integer operations in memory should be abstracted over endianness by LLVM

Reply:
Oops, a nit from my side. You are right Rob!